SpokeStack/TFLiteWakewordRecognizer.swift

```diff
@@ -0,0 +1,478 @@
+        }
+    }
+    
+    /// MARK: Pipeline control
+    
+    private func process(_ frame: Data) -> Void {
+        self.activeLength += 1
+        if self.context.isSpeech && self.activeLength < self.maxActive {
+            /// Run the current frame through the detector pipeline.
+            /// Activate the pipeline if a keyword phrase was detected.
+            do {
+                /// Decode the FFT outputs into the filter model's input
+                try sample(frame)
+                try analyze()
+                try filter()
+                try encode()
+                let activate = try detect()
+                if activate && !self.context.isActive {
+                    self.context.isActive = true
+                    self.reset()
+                    self.stopStreaming(context: self.context)
+                    self.delegate?.activate()
+                }
+            } catch let error {
+                self.delegate?.didError(error)
+            }
+        } else {
+            /// Continue this wakeword (or external) activation
+            /// for at least the activation minimum,
+            /// until a vad deactivation or timeout
+            if (self.activeLength > self.minActive) && (!self.context.isSpeech || (self.activeLength >= self.maxActive)) {
+                self.context.isActive = false
+                self.reset()
+            }
+        }
+    }
+    
+    /// MARK: Audio processing
+    
+    private func sample(_ data: Data) throws -> Void {
+        
+        /// Preallocate an array of data elements in the frame for use in RMS and sampling
+        let dataElements: Array<Int16> = data.elements()
+        
+        /// Update the rms normalization factors
+        /// Maintain an ewma of the rms signal energy for speech samples
+        if self.rmsAlpha > 0 {
+            self.rmsValue = self.rmsAlpha * SignalProcessing.rms(data, dataElements) + (1 - self.rmsAlpha) * self.rmsValue
+        }
+        
+        /// Process all samples in the frame
+        for d in dataElements {
+            
+            /// Normalize and clip the 16-bit sample to the target rms energy
+            var sample: Float = Float(d) / Float(Int16.max)
+            sample = sample * (self.rmsTarget / self.rmsValue)
+            sample = max(-1.0, min(sample, 1.0))
+            
+            /// Run a pre-emphasis filter to balance high frequencies
+            /// and eliminate any dc energy
+            let currentSample: Float = sample
+            sample -= self.preEmphasis * self.prevSample
+            self.prevSample = currentSample
+            
+            //sampleCollector.append(sample)
+            
+            /// Process the sample
+            /// - write it to the sample sliding window
+            /// - run the remainder of the detection pipleline if speech
+            /// - advance the sample sliding window
+            try self.sampleWindow.write(sample)
+            if self.sampleWindow.isFull {
```

Comment:
This is problematic because returning here will ignore any remaining audio samples in `dataElements` once the sample window fills. This is why the android version calls the subsequent functions inside the sample loop.